cmd/create/cluster/cmd.go

```diff
@@ -2852,6 +2862,31 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// Additional Allowed Principals
+	if cmd.Flags().Changed("additional-allowed-principals") && !isHostedCP {
+		r.Reporter.Errorf("Additional Allowed Principals is supported only for Hosted Control Planes")
+		os.Exit(1)
+	}
+	additionalAllowedPrincipals := args.additionalAllowedPrincipals
+	if isHostedCP && interactive.Enabled() {
```

Comment:
Hmm I'm not a big fan of all the if else though. Would it be possible to use validators? One example to compare is with noProxySlice.

I was thinking of something like:
```
aapInputs, err := interactive.GetString(interactive.Input{
			Question: "Additional Allowed Principal ARNs",
			Help:     cmd.Flags().Lookup("additional-allowed-principals").Usage,
			Default:  strings.Join(additionalAllowedPrincipals, ","),
			Validators: [][]interactive.Validator{ValidateAdditionalAllowedPrincipals}
		})
additionalAllowedPrincipals = helper.HandleEmptyStringOnSlice(strings.Split(aapInputs, ","))
```

Reply:
I think the noProxy is validation/validators are redundant.
If you look at the noProxy code they have similar [if slice is populated validation checks ](https://github.com/openshift/rosa/blob/v1.2.40/cmd/create/cluster/cmd.go#L2798-L2811). I think this is because validation needs to be done in param inputs as well.